### ISSUE TITLE
Several issues with 2.0.0-alpha

### ISSUE THREAD

**Issue by ferrannp** [software-mansion/react-native-screens#153]
The branch implementing this can be found https://github.com/ferrannp/fithero/pull/148.

Basically is replacing all the `createStackNavigator` for `createNativeStackNavigator`.

1. On Android, when I have to native stacks, when I navigate to one screen and I go back, I get a red screen:

![red-screen](https://user-images.githubusercontent.com/774577/64619524-85a89b80-d3e2-11e9-9111-0db87fab2397.gif)

2. On Android the tabs are flickering:

![tabs-flicker](https://user-images.githubusercontent.com/774577/64619536-8ccfa980-d3e2-11e9-82a0-e65eb9c4280f.gif)

3. On Android the header is not respecting the content but tabs are ok:

![5f9d837c-4bb4-478b-ad2f-5cae0a5cb0a2](https://user-images.githubusercontent.com/774577/64619586-a113a680-d3e2-11e9-9fe7-09af00b08b36.jpeg)

4. On iOS the header is respecting the content but not the tabs:

![Simulator Screen Shot - iPhone X - 2019-09-10 at 15 50 20](https://user-images.githubusercontent.com/774577/64619645-b4267680-d3e2-11e9-82ab-cfd178d0a613.png)

5. On `mode: modal` there is no header anymore (only on iOS, on Android there is header) and I cannot close the modal using gestures either:

![ios-modal](https://user-images.githubusercontent.com/774577/64619986-55153180-d3e3-11e9-996a-63b713fa3696.gif)

![android-modal](https://user-images.githubusercontent.com/774577/64620079-842ba300-d3e3-11e9-92ca-9b2c5f5abfde.jpeg)

6. On iOS the FAB from `react-native-paper` button is not there anymore. On Android is ok:

Native stack:

![Simulator Screen Shot - iPhone X - 2019-09-10 at 15 52 54](https://user-images.githubusercontent.com/774577/64619846-10899600-d3e3-11e9-809b-66bdb7c2df01.png)

Normal stack:

![Simulator Screen Shot - iPhone X - 2019-09-10 at 15 55 14](https://user-images.githubusercontent.com/774577/64620029-65c5a780-d3e3-11e9-94a3-157225169a1b.png)

7. That can be ok but `headerTitle` does not accept a string anymore so we need to actually come up with a Text component that looks the same as the original.

8. As you see on the screens, I cannot remove the elevation & shadow on toolbar anymore. I used to do it with this code:

```js
headerStyle: {
  elevation: 0,
  backgroundColor: theme.colors.toolbar,
  borderBottomColor: theme.colors.toolbar,
},
```

Not working anymore though.

**Comment by ghondar** [comment]
btw work in progress... https://github.com/kmagiera/react-native-screens#using-native-stack-navigator

**Comment by ferrannp** [comment]
Hey @kmagiera, here the second round of testing after #184 as we discussed. New PR https://github.com/ferrannp/fithero/pull/148.

## iOS

1. I cannot swipe back to the previous screen (gesture)

2. The right buttons on the Toolbar have more padding on the right side that they should. On Android is ok:

<img width="363" alt="Screenshot 2019-10-19 at 14 35 11" src="https://user-images.githubusercontent.com/774577/67145369-6f171f00-f281-11e9-8eaa-790570053651.png">

## Android

1. I am still getting the `removeView` crash if the screen uses a ViewPager. I am using https://github.com/madhu314/react-native-tabbed-view-pager-android which uses React Native Android ViewPager. Also the History tab is empty, it should not be empty. See:

![tabs](https://user-images.githubusercontent.com/774577/67145509-6410be80-f282-11e9-8355-3cb6ae984267.gif)

## Both platforms

1. If you do `headerTitle: () => 'Title'` it crashes. It expects you to wrap it on a `Text` component. But this used to work and user is not force to provide its own styled Text on Android and iOS.

<img width="342" alt="Screenshot 2019-10-19 at 14 57 58" src="https://user-images.githubusercontent.com/774577/67145321-f1ebaa00-f280-11e9-9fd3-94da271dd406.png">

2. I am still not able to remove the borderBottom on iOS and the elevation on Android for the Toolbar:

<img width="363" alt="Screenshot 2019-10-19 at 14 35 11" src="https://user-images.githubusercontent.com/774577/67145351-4f7ff680-f281-11e9-9da2-056bfb279a51.png">

<img width="341" alt="Screenshot 2019-10-19 at 15 02 45" src="https://user-images.githubusercontent.com/774577/67145387-86eea300-f281-11e9-87eb-0affef1884de.png">

The code is:

```js
  headerStyle: {
    elevation: 0,
    backgroundColor: theme.colors.toolbar,
    borderBottomColor: theme.colors.toolbar,
  },
```

**Comment by kmagiera** [comment]
Thanks for checking again!

As for hiding shadow since native headers are not "just RN views" they can no longer be styled this way. I exposed a native stack property called "hideShadow" such that it can now be used from the react-navigation binding: https://github.com/kmagiera/react-native-screens/pull/192

As we discussed on github flickering on android is due to the fact shadows are visible on the navigation bar. If you use the property I mentioned to hide shadow you should no longer observe it. The root cause of why you see it flicker when shadows are ON is in react-native-paper library and needs to be addressed there. In short react-native-paper does not use update alpha compositing when animating between screens. If you turn shadows ON with non native stack you will see it flicker as well.

**Comment by kmagiera** [comment]
As for the back swipe I noticed it was broken in some version but checked recent master and seem to be working there, so chances are one of Janic's PRs we merged in the meantime fixes this

**Comment by tobiaslins** [comment]
@kmagiera What about the headerbar on `mode: 'modal'` for iOS. It's missing for me aswell? Backswipe on modal is working for me!
Thanks again for your work! Do you have opencollective? 🤗

**Comment by ferrannp** [comment]
I'm gonna give it another round of testing when this is merged too https://github.com/kmagiera/react-native-screens/pull/197.

So far with that fix, the only issue on iOS I see is the extra right padding on the toolbar. I need to test Android.

By the way @kmagiera another thing that does not work properly with `react-navigation` and this is the `persistenceKey`. It will persist but not all the controllers so you won't be able to go back :D. Any ideas on how we can fix this?

**Comment by satya164** [comment]
Hey @kmagiera, maybe you can tell me a bit more about what you mean by "In short react-native-paper does not use update alpha compositing when animating between screens"?

**Comment by kmagiera** [comment]
Hey @satya164 – you can check this snack here https://snack.expo.io/@kzzzf/sadistic-chips

You'll have to run this on android. Then when you switch between tabs you will see the button flashing for a fraction of second. If you record the screen and see this in slow motion you will notice that it is due to the button shadows (elevation) not respecting opacity properly while the screen is animated from opacity 1 to 0 (and the one that animates in from 0 to 1). This effect is due to the performance optimization RN views use, it's been documented here https://www.facebook.com/notes/andy-street/react-native-android-offscreen-alpha-compositing/10153949352079590/

The fix would be to set https://facebook.github.io/react-native/docs/view.html#needsoffscreenalphacompositing to `true` for the view that animates opacity but you only want to do that for the duration of animation. For example, react-native screens does that by default when transitioning between screens.

**Comment by kmagiera** [comment]
![out-031](https://user-images.githubusercontent.com/726445/67663754-7d8ec600-f966-11e9-91af-8eecf397d676.png)

**Comment by kmagiera** [comment]
This is a mid-animation state above. As on both screens in between you animate have white top it should never be the case that the top of the screen is not white because you animate white into white. The same should be for the bottom of the screen and the bottom should also not change its color for the duration of animation.

**Comment by satya164** [comment]
Makes sense. Thanks for the explanation. I think we'll also need to do this for react-navigation's animations.

One issue I noticed is that when I specify this prop, all of the elevation style is removed from the children views during the time the animation happens. Without this prop it just stays at full opacity during the animation so I guess it's a bit better.

**Comment by wibb36** [comment]
> Hey @tobiaslins thanks for your feedback. I haven't yet decided what to do with the header on modals. I'm leaning towards disabling it on Android too. In native iOS in order to have the navbar there you'd render another stack inside a modal. There is a way to only render plain navbar but then you loose a bunch of styling options that are available when using the whole navigator. I'll keep you updated which route I ended up choosing, if you have any suggestions I'd love to hear them.
> 
> As for the open collective I don't have one, what I'd be more happy about is if you decided to support RN community events I help organize, e.g., you can buy a ticket to [App.js 2020](https://appjs.co) conference or just tell your frends or coworkers about it :)

@kmagiera Have you thought you wanted to do with the header on modal? It's still missing on RNS 2.3.0. I'd collaborate but I'm not an iOS developer. I'd appreciate your help with this.

**Comment by WoLewicki** [comment]
It looks like this issue is concerning #260 only now, so can I close it @ferrannp ?

**Comment by ferrannp** [comment]
Hey @WoLewicki ! Yes let's close this as I need to do another round of tests but if I find another issue, I can open a new one. I think this has been long enough here :).
